internal/decode/decoder.go

```diff
@@ -107,25 +108,26 @@ func directHTTPDecoder(u *url.URL) Func {
 }
 
 // Uses the files passed in the args and looks for the specified schemaFile to parse the YAML.
-func unmarshalAsYAMLOrSchemaWithFile(data []byte, out interface{}, args []string) (bool, error) {
+func unmarshalAsYAMLOrSchemaWithFile(data []byte, out interface{}, filename string) (bool, error) {
 	if strings.Contains(string(data), "schemaFile:") && !strings.Contains(string(data), "schema:") {
 		if err := yaml.Unmarshal(data, out); err != nil {
 			return false, err
 		}
-		schema := out.(*validationfile.ValidationFile)
-
-		for _, arg := range args {
-			// Check if the file specified in schemaFile is passed as an arguement.
-			if strings.Contains(arg, schema.SchemaFile) {
-				file, err := os.Open(arg)
-				if err != nil {
-					return false, err
-				}
-				data, err = io.ReadAll(file)
-				if err != nil {
-					return false, err
-				}
-			}
+		validationFile := out.(*validationfile.ValidationFile)
+
+		// Need to join the original filepath with the requested filepath
+		// to construct the path to the referenced schema file.
+		// NOTE: This does not allow for yaml files to transitively reference
```

Comment:
Add a TODO to enable this. Also, we should validate that the file specified is at, or below, the directory path so it can't be used to open arbitrary files